dm/dm/ctl/master/source_table_schema.go

```diff
@@ -57,27 +57,54 @@ func newSourceTableSchemaUpdateCmd() *cobra.Command {
 			if len(args) < 3 {
 				return cmd.Help()
 			}
-			taskName := common.GetTaskNameFromArgOrFile(args[0])
-			sources, err := common.GetSourceArgs(cmd)
+
+			var (
+				taskName, database, table           string
+				sources                             []string
+				schemaContent                       []byte
+				flush, sync, fromSource, fromTarget bool
+				err                                 error
+			)
+
+			fromSource, err = cmd.Flags().GetBool("from-source")
 			if err != nil {
 				return err
 			}
-			database := args[1]
-			table := args[2]
-			schemaFile := args[3]
-			schemaContent, err := common.GetFileContent(schemaFile)
+			fromTarget, err = cmd.Flags().GetBool("from-target")
 			if err != nil {
 				return err
 			}
-			flush, err := cmd.Flags().GetBool("flush")
+
+			if !fromSource && !fromTarget && len(args) < 4 {
```

Comment:
do we need also check `fromSource && fromTarget` too?

Reply:
yes, i will add check.